source/kraken/maintenance_worker.cpp

```diff
@@ -355,13 +359,28 @@ void MaintenanceWorker::init_rabbitmq() {
     queue_name_task = (boost::format("%s_task") % queue_name).str();
     queue_name_rt = (boost::format("%s_rt") % queue_name).str();
 
-    // connection through URI or classic method
-    if (broker_uri) {
-        LOG4CPLUS_INFO(logger, boost::format("connection to rabbitmq with uri: %s") % *broker_uri);
-        channel = AmqpClient::Channel::CreateFromUri(*broker_uri);
-    } else {
-        LOG4CPLUS_INFO(logger, boost::format("connection to rabbitmq: %s@%s:%s/%s") % username % host % port % vhost);
-        channel = AmqpClient::Channel::Create(host, port, username, password, vhost);
+    if (protocol == "amqp") {
+        // connection through URI or classic method
+        if (broker_uri) {
+            LOG4CPLUS_DEBUG(logger, boost::format("connection to rabbitmq with uri: %s") % *broker_uri);
+            channel = AmqpClient::Channel::CreateFromUri(*broker_uri);
+        } else {
+            LOG4CPLUS_DEBUG(logger, boost::format("connection to rabbitmq: %s://%s@%s:%s/%s") % protocol % username
+                                       % host % port % vhost);
+            channel = AmqpClient::Channel::Create(host, port, username, password, vhost);
+        }
+    } else if (protocol == "amqps") {
+        // connection through URI or classic method
+        if (broker_uri) {
+            LOG4CPLUS_DEBUG(logger, boost::format("connection to rabbitmq with uri: %s") % *broker_uri);
+            channel = AmqpClient::Channel::CreateSecureFromUri(*broker_uri, path_to_ca_cert, path_to_client_key,
```

Comment:
To test: How about passing `verify_hostname=false` ?

Reply:
`verify_hostname=true` is fine... but `verify_peer=false` was the key!